File: 3005. Count Elements With Maximum Frequency/maxFrequencyElements.py
from typing import List

class Solution:
  def maxFrequencyElements(self, nums: List[int]) -> int:
    frequencyNums = {}
    maxFrequency = 0
    for num in nums:
      if num not in frequencyNums:
        frequencyNums[num] = 0
      frequencyNums[num] += 1
      maxFrequency = max(maxFrequency, frequencyNums[num])

    maxFreqCount = 0
    for val in frequencyNums.values():
      if val == maxFrequency:
        maxFreqCount += maxFrequency

    return maxFreqCount

nums = [1, 2, 2, 3, 1, 4]
solution = Solution()
print('maxFrequencyElements: ', solution.maxFrequencyElements(nums))

File: 3005. Count Elements With Maximum Frequency/maxFrequencyElements.js
/**
 * @param {number[]} nums
 * @return {number}
 */
const maxFrequencyElements = (nums) => {
  const frequencyNums = {};
  let maxFrequency = 0;

  for (let i = 0; i < nums.length; i++) {
    if (!frequencyNums[nums[i]]) {
      frequencyNums[nums[i]] = 0;
    }

    frequencyNums[nums[i]]++;
    maxFrequency = Math.max(maxFrequency, frequencyNums[nums[i]]);
  }

  let maxFreqCount = 0;
  for (let num in frequencyNums) {
    if (frequencyNums[num] === maxFrequency) {
      maxFreqCount += maxFrequency;
    }
  }

  return maxFreqCount;
};

const nums = [1, 2, 2, 3, 1, 4];
console.log("maxFrequencyElements: ", maxFrequencyElements(nums));
